src/prefect/agent/fargate/agent.py

```diff
@@ -376,6 +376,11 @@ def _parse_kwargs(self, user_kwargs: dict, check_envars: bool = False) -> tuple:
         except (ValueError, SyntaxError):
             pass
 
+        if len(container_defs) > 1:
+            raise ValueError(
+                "Multiple container definitions provided to Fargate agent."
+            )
```

Comment:
Should probably fail if it's len == 0 as well. Perhaps:

```suggestion
        if len(container_defs) != 1:
            raise ValueError(
                "Fargate agent only accepts configuration for a single container definitiont."
            )
```